### ISSUE TITLE
Checking the measures costs and the implications of updating the house value when measures are implemented

### ISSUE THREAD

**Issue by vjcortesa** [averbraeck/housinggame-player#65]
**Checking the measure costs**
There is a misbalance between flood adaptation measures costs, which may affect the cost-benefit ratio for flood protection (learning output of the game):

- **WhereWeMove Green garden (20k) and underground barrel (11k).** Compared with the Sudsbury the green garden is the cheapest measure and the rain water recycling system comparable to the description of the underground barrel is the most expensive.
A solution is to keep the price and improve the description for example with the garden including a retention pond for water, so that its value is not for the green area and increase the value of the underground barrel to perhaps 15k.

- **Water Resistant walls & floors (20k) is as expensive as the green garden (20k)** and when compared to modest house renovations is comparable to the satisfaction cost per point for the player with the highest welfare (21k). When doing house renovations there is the possibility to renewing a room, painting or flooring adding to 1 satisfaction point (from 4k to 21k). 
Is it really this measure as expensive? Should we clarify that its implementation refers to adequating the first floor to get flooded, if such is the outside painting of the house, floors with open pavement with possible and adequating some house areas to get flooded if necessary?

**Checking how difficult and meaningful is to implement the house values increase**, 
We already have flood rates reductions to the house value, so not sure if that helps the implementation.
However, when sells to the bank, the bank should then pay the updated value. 
If we Increase the house value on 5% when implementing a measure gives between 4k (80k houses) and 20k (425k houses). As such, house values increase only makes sense to:
- Green garden (20k)
- Resistant walls & Floors (20k)
- Underground barrel (including recycling) (11k)
- House renovations (only when 2 points) (from 8k to 21k). 
  
![image](https://github.com/user-attachments/assets/230ab95b-2e66-4d7f-9e0d-c90912d60b67)
![image](https://github.com/user-attachments/assets/7d09a8d9-e6a9-469d-b779-5256f453f49e)

**Comment by averbraeck** [comment]
The sandbags are quite difficult to maintain. It would simplify matters and game play a lot if we can change it to have a validity of one round only. In a sense, it would work like a cheap insurance option. But if you want it, you buy it for that round. The dice are rolled, and it might help you to be better protected -- or you might not need it. Next round, you can buy sandbags again. 

I would implement insurance for rain protection in exactly the same way, but as a much more expensive option. You also pay it per round. It carries a validity of one round, but protects you at the highest level of rain damage. If you have money, you can insure or buy sandbags again in the next round. 

This would mean, we have two types of measure types:
1. **Permanent-effect**. They might increase the flood or rain protection level permanently. They might give satisfaction points per round or just once at buying. Examples are green garden, pump, bulkhead, etc.
2. **One-round-effect**. They might increase the flood or rain protection level for the current round only. They might give satisfaction points for this round only. Examples are sandbags, insurance, house improvements, personal improvements.

Each measure type is categorized as being **flood related** or **personal**. 

Each measure type has a **fixed cost** or a **percentage of income cost**. 

This is an extremely flexible way of implementing the measures. The combinations cover all kinds of situations.

**Comment by averbraeck** [comment]
About the house price increase: if I add a field that indicates how much the house price increases as a result of the implementation of the measure, I think we solved it. Implementation of house price increase will come later, though. 

Usually, measures increase the house price with 25-50% of the cost of that measure. And there can be exceptions. A green garden will (unfortunately) not add to the house price, where impregnated walls might increase the house price with 50% of the cost of the measure.

**Comment by vjcortesa** [comment]
> The sandbags are quite difficult to maintain. It would simplify matters and game play a lot if we can change it to have a validity of one round only. In a sense, it would work like a cheap insurance option. But if you want it, you buy it for that round. The dice are rolled, and it might help you to be better protected -- or you might not need it. Next round, you can buy sandbags again.
> 
> I would implement insurance for rain protection in exactly the same way, but as a much more expensive option. You also pay it per round. It carries a validity of one round, but protects you at the highest level of rain damage. If you have money, you can insure or buy sandbags again in the next round.
> 
> This would mean, we have two types of measure types:
> 
> 1. **Permanent-effect**. They might increase the flood or rain protection level permanently. They might give satisfaction points per round or just once at buying. Examples are green garden, pump, bulkhead, etc.
> 2. **One-round-effect**. They might increase the flood or rain protection level for the current round only. They might give satisfaction points for this round only. Examples are sandbags, insurance, house improvements, personal improvements.
> 
> Each measure type is categorized as being **flood related** or **personal**.
> 
> Each measure type has a **fixed cost** or a **percentage of income cost**.
> 
> This is an extremely flexible way of implementing the measures. The combinations cover all kinds of situations.

**Comment by vjcortesa** [comment]
I like the categorization of measures according to their effect but would not put sandbags (emergency measures) and insurance (recovery) in the same category because, behaviourally, they are very different. 

Below is my proposal, also considering some tweaks that allow to offer active participation as part of the insurance or a separate measure, for example.

Permanent-effect: 
- They might permanently increase the flood or rain protection level. 
- They might give satisfaction points per round or just once when buying. Examples are green garden, pump, bulkhead, etc.
- They might increase the house value.
- They might decrease the insurance value per round.

One-round-effect. 
- They might only increase the flood or rain protection level for the current round. 
- They might only give satisfaction points for this round (personal and house improvements). 
- They might increase the house value: house improvements but also active participation (allow to join emergency training, real-time updates to place sandbags earlier and may even give some personal satisfaction)
- Their value might change depending on what other measures are implemented (insurance if players implement permanent measures or permanent until use.
- Their effect might improve with, for example, some effort (e.g., participating in emergency training or guidance on how to maintain them).

Permanent until-use. 
- Their effect might improve when you know how to use them (sandbags are the only type of emergency measure or commit to active participation). Residents need to know that they need to be placed and give protection for longer to players who do not have money to buy other measures while still being temporary measures.

**Comment by averbraeck** [comment]
I fully agree with the differences in **behavioral** effects, and that they fall under completely **different categories**. Implementation-wise, however, they are exactly the same. You pay, and the pluvial or fluvial damage is decreased in that round. I really want to kick out the until-use option, since it is a nightmare to implement.

**Comment by averbraeck** [comment]
Combined measures with additional effects: yes in reality; for me, a no-go in the game. Too complex to explain or remember; too complex to implement in the database; too complex to implement in the player application (debugging the correct implementation of combined effects will be hard).

**Comment by averbraeck** [comment]
"They might decrease the insurance value per round". You probably mean the insurance premium aka the insurance cost? Not the insured value I suppose. Again, too complex (also in reality -- insurance companies typically do not decrease the annual cost based on measures you took to protect your house right now). The way I see it for the next version of the game, you pay x% of the house value per year for insurance, where you can set x in the measure type.

For later or other games, it might be interesting to look at combined effects, though, such as a lower insurance price when certain measures are implemented.

**Comment by averbraeck** [comment]
I do not see how we can implement the effects of effort or training in the game for measures, nor for the need for extra house space. Interesting for later extensions or completely new games, not for the current game.

**Comment by averbraeck** [comment]
So, for the next iteration of the game, I will go with the changes that will be worked out in detail in https://github.com/averbraeck/housinggame-common/issues/26.

**Comment by averbraeck** [comment]
"I did not understand why a green garden does not increase the house value": just a personal observation from buying and selling houses myself. The type of garden is typically something that is so dependent on taste, that the type of garden cannot easily be used to add value to the house. Most of my neighbors completely changed the garden after buying the house (and so did we -- every time). 

In the game, however, we can do with this whatever we want. If the green garden costs 20 kE, we can add 5kE or 10kE or 20 kE to the house value if we want. That is up to you. I was mentioning my observations in real life.

**Comment by averbraeck** [comment]
All effects are implemented. Including one-time measures such as sandbags, and one-round measures such as insurance.
